src/test/Directory.Build.props

```diff
@@ -12,4 +12,14 @@
     <TestInfraTargetFramework>netcoreapp3.0</TestInfraTargetFramework>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <TargetOSTrait Condition="'$(OSGroup)' == 'Windows_NT'">nonwindowstests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(OSGroup)' == 'Linux'">nonlinuxtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(OSGroup)' == 'OSX'">nonosxtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(OSGroup)' == 'FreeBSD'">nonfreebsdtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(OSGroup)' == 'NetBSD'">nonnetbsdtests</TargetOSTrait>
+
+    <TestRunnerAdditionalArguments Condition="'$(TargetOSTrait)' != ''">-notrait category=$(TargetOSTrait)</TestRunnerAdditionalArguments>
+  </PropertyGroup>
```

Comment:
@dagood, IMO this is bit of an oddity in arcade SDK, that it provides everything required to use `PlatformSpecificAttribute`, except for the `TargetOSTrait` (this property group), which is defined in [corefx specific targets](https://github.com/dotnet/arcade/blob/5b66439eb4e396f105548150acffbaf062d9db76/src/Microsoft.DotNet.CoreFxTesting/build/core/test/Test.targets#L23).

Reply:
Opened dotnet/arcade#4105.